plugins/module_utils/common.py

```diff
@@ -520,8 +521,15 @@ def remove_aliases(self):
                     if alias in self.params:
                         self.params.pop(alias)
 
-    def load_resource_definitions(self, src):
+    def load_resource_definitions(self, src, module=None):
         """Load the requested src path"""
+        if module and (
+            src.startswith("https://")
+            or src.startswith("http://")
+            or src.startswith("ftp://")
+        ):
+            src = fetch_file(module, src)
```

Comment:
I think using `fetch_file()` here is potentially a problem. This will use the same `validate_certs` value that's used for connecting to the cluster, and we have no control over whether or not it does this. I would suggest that we use `open_url()` instead, or just use `requests` since we are already dependent on this anyways.